front/app/components/ConsentManager/utils.ts

```diff
@@ -0,0 +1,105 @@
+// utils
+import { isNilOrError, NilOrError } from 'utils/helperUtils';
+
+// cookies
+import {
+  allCategories,
+  getDestinationConfigs,
+  isDestinationActive,
+  IDestinationConfig,
+} from './destinations';
+import { IConsentCookie } from './consent';
+
+// typings
+import { IAppConfigurationData } from 'services/appConfiguration';
+import { CategorizedDestinations } from './typings';
+import { TAuthUser } from 'hooks/useAuthUser';
+
+export const getCategory = (
+  tenant: IAppConfigurationData,
+  destinationConfig: IDestinationConfig
+) => {
+  return typeof destinationConfig.category === 'function'
+    ? destinationConfig.category(tenant)
+    : destinationConfig.category;
+};
+
+export const categorizeDestinations = (
+  appConfiguration: IAppConfigurationData | NilOrError,
+  destinations: IDestinationConfig[]
+): CategorizedDestinations => {
+  const output: CategorizedDestinations = {
+    analytics: [],
+    advertising: [],
+    functional: [],
+  };
+
+  if (isNilOrError(appConfiguration)) {
+    return output;
+  }
+
+  destinations.forEach((destinationConfig) => {
+    const category = getCategory(appConfiguration, destinationConfig);
+    output[category].push(destinationConfig.key);
+  });
+
+  return output;
+};
+
+export const getActiveDestinations = (
+  appConfiguration: IAppConfigurationData | NilOrError,
+  authUser: TAuthUser
+): IDestinationConfig[] => {
+  if (isNilOrError(appConfiguration)) return [];
```

Comment:
I agree, but this is how the code was already, and there's another 100 things I would love to improve about it- which I resisted to do to avoid ending up in a rabbit hole